src/components/Transactions.js

```diff
@@ -27,15 +28,38 @@ class Transactions extends React.Component {
       loaded: false,
       hasAfter: false,
       hasBefore: false,
+      queryParams: this.obtainQueryParams(),
     }
   }
 
   componentDidMount() {
-    this.getData(true, null, null, '');
+    this.getData(this.state.queryParams);
 
     WebSocketHandler.on('network', this.handleWebsocket);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    const queryParams = this.obtainQueryParams();
+
+    // Do we have new URL params?
+    if (JSON.stringify(this.state.queryParams) !== JSON.stringify(queryParams)) {
+      // Fetch new data, unless query params were cleared and we were already in the most recent page
+      if (queryParams.hash || this.state.hasBefore) {
+        this.getData(queryParams);
+      }
+      this.setState({ queryParams: queryParams });
```

Comment:
Just update it to `this.setState({ queryParams });` to keep the same code style, please.